packages/runtimes/js/src/JSRuntime.js

```diff
@@ -64,6 +65,14 @@ function getLoaders(
   return null;
 }
 
+let bundleAsyncDependencies = new WeakMap<
```

Comment:
This isn't the dependencies themselves, it's their resolutions. If we cache the dependencies themselves, we can even warm this cache during the initial traversal when determining them for loaders here:

https://github.com/parcel-bundler/parcel/blob/a4cf3e958ea1136b4829eb93615fba5328213471/packages/runtimes/js/src/JSRuntime.js#L82